Relay board

joelleg123 opened the pull request:
[Project Proposal](https://docs.google.com/document/d/1ex50tJbJoOFx4_PS8z7fkQgzuIM5GV6kWhBGIFeKA9E/edit?usp=sharing)

joelleg123 committed:
Add relay board

joelleg123 committed:
Valve Ignition Relay - ready for review 1

joelleg123 committed:
Replaced MOSFETs with optoisolators

joelleg123 committed:
Added IO Expander and moved dip switch

joelleg123 committed:
Added NC to extra pins

joelleg123 committed:
Added 4 new signals to the microcontroller

joelleg123 committed:
Changed Folder Name maybe?

joelleg123 committed:
rename relay board folder

StarlightDescender reviewed (dismissed):
1. Swap the pin order of J6 so the pin numbers of J5 and J6 match
2. What's the current limit on F2? What are the part numbers for both of the fuses? Also what's the reason for F3 being sized for 200A? That's really big...
3. Is there a reason we have a 6 channel DIP switch? my understanding is that gives us 64 board IDs, is 16 (4 channels, 2^4 -> 16)
4. aaaaaa global labels 😭 (You should use "Hierarchical Labels", they better communicate the connections between subsheets)
5. 
![image](https://github.com/user-attachments/assets/f7412cfd-e295-4b28-ac50-e68161db517f)
GND should point down
6. A few footprints missing/dependent on the specific part?

joelleg123 committed:
Changed to Hierarchical labels and added reverse voltage protection. Still need to do resistor math for voltage protection.

joelleg123 committed:
Finished Reverse Voltage Math, added labels

StarlightDescender reviewed (changes requested):
1. Missing datasheets for some of the diodes, optocoupler, relay, etc
2. The maximum current through the collector of U5 is 50mA, but with only the coil resistance of U3 you get 12/160 = 75mA, so you'll need a resistor between pin 2 of U3 and pin 4 of U5. Same with U4/U6 (but the coil resistance is 86 +- 10% rather than 160, so the resistor size will be different).
3. Moreso because I don't know, what's the purpose between the schottky diodes on the voltage sensing lines (VSENSE_OUT1 and VSENSE_OUT2)?
4. The hierarchical labels on the root sheet aren't actually doing anything, those can be deleted
5. Change the custom footprint library's name to 'relay_board' or something more descriptive than 'footprints'
6. footprints for the 1N5819W diode are wrong, from digikey product page it should be SOD-123F. Check the footprints for the optocouplers as well.

joelleg123 committed:
requested changes 1

joelleg123 commented:
1. Datasheets were added (hopefully).
2. Changed the optoisolator to accommodate current so that we do not reduce relay current.
3. If I am understanding, those lines are connected to battery power via the relays, therefore they need to be protected the same way the battery does? 
4. Done!
5. Done!
6. Fixed :))

Joe-Joe-Joe-Joe committed:
fix symbols; routing v1

Joe-Joe-Joe-Joe committed:
lowpass current sense; move power traces to FCu

Joe-Joe-Joe-Joe committed:
Add untracked files; fix excess sym libs

Joe-Joe-Joe-Joe committed:
Silkscreen extras

Joe-Joe-Joe-Joe committed:
Push JSTs to board edge

StarlightDescender reviewed (changes requested):
- D6/7 should be DNP, D8/9 should be placed (for the current ball valve actuators they are the same because of the internals, but this doesn't work for other actuators where a flyback diode is necessary)
- Adjust cutoff frequency of current sense amp outputs to something like 500Hz (Nyquist says that fc should be at most half the sampling frequency)
- P1/G1 traces need to be much thicker to be at least somewhat okay with 24A. P2/G2 are fine as they are though.

StarlightDescender committed:
tidy up traces, add board name silkscreen

StarlightDescender committed:
add team logo

Pdada1 reviewed (changes requested):
<img width="648" height="249" alt="image" src="https://github.com/user-attachments/assets/3b558ac3-249d-4068-9b36-d2d27adc9748" />
This resistor is calculated wrong. If you have this connected to the 5V top rail wont you hit rail way before your 200mA of current.Your max is 83mA before rail. 

<img width="241" height="231" alt="image" src="https://github.com/user-attachments/assets/6cb24ac2-83c6-405e-9322-e41870504c86" />

Show cutoff frequency of the calculation. 

<img width="1165" height="547" alt="image" src="https://github.com/user-attachments/assets/cd13b13c-d72d-4a52-b0a3-e8cfe514a174" />
Talked with Jason about this, I think it's fair to not have reverse polarity protection on this as long as we change the connector on this board to be clocked so that we retain the protection that comes from power in from Reverse Polarity Board. 

<img width="442" height="225" alt="image" src="https://github.com/user-attachments/assets/7951a276-3fb6-48f1-8993-569a8e0c967e" />

I think the 40V here is an error

<img width="589" height="609" alt="image" src="https://github.com/user-attachments/assets/166f7d39-614a-4212-95b5-ca0d785a1449" />

Nit but move the line so it isn't hidden by box

<img width="282" height="196" alt="image" src="https://github.com/user-attachments/assets/55c82e12-6b5b-46ba-8346-138aee7d56eb" />

What's the context for this calculation

<img width="781" height="598" alt="image" src="https://github.com/user-attachments/assets/be4c4ab3-93ef-49e5-ae86-50e145b7030b" />
Values in here don't seem to fully match datasheet. Although idk if putting more current into the LED is necessarily bad, just means we open the BJT a lot more as far as I can tell since max IF = 50mA

<img width="1255" height="747" alt="image" src="https://github.com/user-attachments/assets/9b07bf6a-7a41-40ec-be1f-3a1c7d7c9c60" />
Might be nice to be able to probe this common line here to tell in a relay failure if the first relay is open or closed

Otherwise schematic looks good to me. 
Will review footprints and layout in next review

Pdada1 reviewed (changes requested):
<img width="840" height="472" alt="image" src="https://github.com/user-attachments/assets/75c22e39-5b90-40ab-9e06-41c52c5f0c91" />
This part of the pour is pretty thin for a 30A
<img width="370" height="481" alt="image" src="https://github.com/user-attachments/assets/e967e6b7-6ca1-41b7-9ce4-51d03d276839" />
On this same issue, the CANH/L lines mean only the small area behind the pads are what connects the traces. Can the CAN lines run on the power/GND plane instead so that we have a lot more pour width

Idk, just in general the power traces look too thin to be able to handle the 25ish amps running through it. Even at 60C temp rise, the width is not enough. For this board probably 2oz/ft copper thickness will be required in order to reduce trace width req
